src/account/operations/transfer/high_level/burn_native_token.rs

```diff
@@ -0,0 +1,250 @@
+                        byte_cost_config,
+                        foundry_id.serial_number(),
+                        token_id.token_tag(),
+                        TokenScheme::Simple(SimpleTokenScheme::new(
+                            *foundry_simple_ts.minted_tokens(),
+                            foundry_simple_ts.melted_tokens() + burn_token_amount,
+                            *foundry_simple_ts.maximum_supply(),
+                        )?),
+                    )?
+                    .add_unlock_condition(UnlockCondition::ImmutableAliasAddress(
+                        ImmutableAliasAddressUnlockCondition::new(AliasAddress::from(alias_id)),
+                    ))
+                    .finish()?,
+                ),
+            ];
+            self.send(outputs, options).await
+        } else {
+            unreachable!("We checked if it's an alias output before")
+        }
+    }
+
+    /// Function to burn native tokens without foundry
+    pub async fn burn_native_token_without_foundry(
+        &self,
+        native_token: (TokenId, U256),
+        options: Option<TransferOptions>,
+    ) -> crate::Result<TransferResult> {
+        log::debug!("[TRANSFER] burn_native_token_without_foundry");
+        let token_id = native_token.0;
+        let burn_token_amount = native_token.1;
+
+        let (custom_inputs, outputs) = self.select_native_token_output(token_id, burn_token_amount).await?;
+
+        let options = match options {
+            Some(mut options) => {
+                options.custom_inputs.replace(custom_inputs);
+                options.allow_burning = true;
+                Some(options)
+            }
+            None => Some(TransferOptions {
+                custom_inputs: Some(custom_inputs),
+                allow_burning: true,
+                ..Default::default()
+            }),
+        };
+
+        self.send(outputs, options).await
+    }
+
+    async fn select_native_token_output(
+        &self,
+        token_id: TokenId,
+        burn_token_amount: U256,
+    ) -> crate::Result<(Vec<OutputId>, Vec<Output>)> {
+        let account = self.read().await;
+        let mut inputs_and_outputs = Vec::new();
+        for (output_id, output_data) in account.unspent_outputs().iter() {
+            match output_data.output {
+                Output::Alias(_) | Output::Basic(_) | Output::Nft(_) => {
+                    if let Some(native_tokens) = output_data.output.native_tokens() {
+                        let mut amount = U256::from(0);
+                        let mut not_to_be_burnt_native_tokens = Vec::new();
+                        for native_token in native_tokens.iter() {
+                            if *native_token.token_id() == token_id {
+                                amount += *native_token.amount();
+                            } else {
+                                not_to_be_burnt_native_tokens.push(native_token);
```

Comment:
Should probably use `add_native_token()` https://github.com/iotaledger/bee/blob/146ce1011fd76f3d9f07390b002b29dd6762daf3/bee-block/src/output/native_token.rs#L70